feat: discovery modules from transports should be added (#510)
libp2p/js-libp2p: doc/CONFIGURATION.md

- [js-libp2p-kad-dht](https://github.com/libp2p/js-libp2p-kad-dht)
- [js-libp2p-webrtc-star](https://github.com/libp2p/js-libp2p-webrtc-star)

> Take into consideration that `peer-discovery` services within transports (such as `js-libp2p-webrtc-star`) are automatically gathered from the `transport` and do not need to be added in the discovery modules. However, these transports can also be configured and disabled as the other ones.

Reviewer: ```suggestion
**Note**: `peer-discovery` services within transports (such as `js-libp2p-webrtc-star`) are automatically gathered from the `transport`, via it's `discovery` property. As such, they do not need to be added in the discovery modules. However, these transports can also be configured and disabled as the other ones.
```
